src/Catalyst.Core.Lib/Validators/TransactionValidator.cs

```diff
@@ -0,0 +1,108 @@
+*/
+
+#endregion
+
+using System.Linq;
+using Catalyst.Abstractions.Cryptography;
+using Catalyst.Abstractions.Validators;
+using Catalyst.Protocol.Network;
+using Catalyst.Protocol.Wire;
+using Google.Protobuf;
+using Serilog;
+
+namespace Catalyst.Core.Lib.Validators
+{
+    public sealed class TransactionValidator : ITransactionValidator
+    {
+        private readonly ILogger _logger;
+        private readonly ICryptoContext _cryptoContext;
+
+        public TransactionValidator(ILogger logger,
+            ICryptoContext cryptoContext)
+        {
+            _cryptoContext = cryptoContext;
+            _logger = logger;
+        }
+
+        public bool ValidateTransaction(TransactionBroadcast transactionBroadcast, NetworkType networkType)
+        {
+            return ValidateTransactionFields(transactionBroadcast)
+             && CheckContractInputFields(transactionBroadcast)
+             && CheckCfEntries(transactionBroadcast)
+             && CheckStEntries(transactionBroadcast)
+             && ValidateTransactionSignature(transactionBroadcast, networkType);
+        }
+
+        private bool CheckContractInputFields(TransactionBroadcast transactionBroadcast)
+        {
+            // @TODO DO SOMETHING
+            return true;
+        }
+
+        private bool ValidateTransactionFields(TransactionBroadcast transactionBroadcast)
+        {
+            // @TODO DO SOMETHING
+            return true;
+        }
+
+        private bool ValidateTransactionSignature(TransactionBroadcast transactionBroadcast, NetworkType networkType)
+        {
+            if (transactionBroadcast.Signature.RawBytes == ByteString.Empty)
+            {
+                _logger.Error("Transaction signature is null");
+                return false;
+            }
+
+            var transactionSignature = _cryptoContext.GetSignatureFromBytes(transactionBroadcast.Signature.RawBytes.ToByteArray(),
+                transactionBroadcast.PublicEntries.First().Base.SenderPublicKey.ToByteArray());
+            var transactionWithoutSig = transactionBroadcast.Clone();
+            transactionWithoutSig.Signature = null;
+
+            if (!_cryptoContext.Verify(transactionSignature, transactionWithoutSig.ToByteArray(), transactionBroadcast.Signature.SigningContext.ToByteArray()))
+            {
+                return true;
+            }
+            
+            _logger.Information(
+                "Transaction Signature {signature} invalid.",
+                transactionSignature);
+            return false;
+        }
+
+        private bool CheckCfEntries(TransactionBroadcast transactionBroadcast)
+        {
+            // @TODO DO SOMETHING
+            return true;
+        }
+
+        private bool CheckStEntries(TransactionBroadcast transactionBroadcast)
+        {
+            // @TODO DO SOMETHING
+            return true;
+        }
+
+        private bool CheckKeySize(ByteString publicKey)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private method 'CheckKeySize'.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)